src/leonardo_battery_guard/scripts/battery_guard.py

```diff
@@ -0,0 +1,79 @@
+from kobuki_msgs.msg import AutoDockingAction, AutoDockingGoal
+from actionlib_msgs.msg import GoalStatus
+from diagnostic_msgs.msg import DiagnosticArray
+
+
+class AutoDocking(object):
+    def __init__(self):
+        self._ros_node = rospy.init_node("dock_drive_client", anonymous=True)
+        self._client = actionlib.SimpleActionClient("dock_drive_action", AutoDockingAction)
+        self._diagnostic_agg_sub = rospy.Subscriber("/diagnostics_agg", DiagnosticArray, self._listen_batteries)
+        self._doing_docking = False
+        self.BATTERY_THRESHOLD = rospy.get_param("~battery_threshold", 10)
+
+    def _doneCb(self, status, result):
+        """Callback that prints the action goal status and change the state of _doing_docking attribute. 
+        This function is called when the action goal is reached."""
+        if status == GoalStatus.PENDING: state="PENDING"
+        elif status == GoalStatus.ACTIVE: state="ACTIVE"
+        elif status == GoalStatus.PREEMPTED: state="PREEMPTED"
+        elif status == GoalStatus.SUCCEEDED: state="SUCCEEDED"
+        elif status == GoalStatus.ABORTED: state="ABORTED"
+        elif status == GoalStatus.REJECTED: state="REJECTED"
+        elif status == GoalStatus.PREEMPTING: state="PREEMPTING"
+        elif status == GoalStatus.RECALLING: state="RECALLING"
+        elif status == GoalStatus.RECALLED: state="RECALLED"
+        elif status == GoalStatus.LOST: state="LOST"
+        rospy.logdebug("Result - [ActionServer: " + state + "]: " + result.text)
+        self._doing_docking = False
+
+    def _feedbackCb(self, feedback):
+        """Callback that just prints the action feedback."""
+        rospy.logdebug("Feedback: [DockDrive: " + feedback.state + "]: " + feedback.text)
+
+    def _go_dock(self):
+        """Runs auto docking routine for a turtlebot."""
+        if rospy.is_shutdown(): 
+            return        
+        goal = AutoDockingGoal()
+        self._client.send_goal(goal, done_cb=self._doneCb, feedback_cb=self._feedbackCb)
+        rospy.logdebug("Goal: Sent.")
+        rospy.on_shutdown(self._client.cancel_goal)
+        self._doing_docking = False
+    
+    def _listen_batteries(self, data):
+        """Callback that checks the batteries charge and triggers the auto docking routine if the charge is low."""
+        batteries_names = ["/Power System/Laptop Battery", "/Power System/Battery"]
+        batteries_values = [element.values for element in data.status if element.name in batteries_names]
+        laptop_battery_percentage = float(filter(lambda x: x.key == "Percentage (%)", batteries_values[0])[0].value)
+        kobuki_battery_percentage = float(filter(lambda x: x.key == "Percent", batteries_values[1])[0].value)
+        
+        if self._doing_docking is False:
+            if kobuki_battery_percentage < self.BATTERY_THRESHOLD or laptop_battery_percentage < self.BATTERY_THRESHOLD:
+                self._go_dock()
+                self._doing_docking = True
```

Comment:
What happens here if the robot is docked and charging but the battery level is below the threshold? Does the docking action get triggered? Have you tested this case?